File: leetcode/1732. Find the Highest Altitude/main.go
package main

func largestAltitude(gain []int) int {
	maxAltitude, curAltitude := 0, 0

	for _, gain := range gain {
		curAltitude += gain
		if curAltitude > maxAltitude {
			maxAltitude = curAltitude
		}
	}

	return maxAltitude
}

File: leetcode/1732. Find the Highest Altitude/main.rs
impl Solution {
    pub fn largest_altitude(gains: Vec<i32>) -> i32 {
        let (mut cur_alt, mut max_alt) = (0, 0);

        for gain in &gains {
            cur_alt = cur_alt + gain;
            if cur_alt > max_alt {
                max_alt = cur_alt
            }
        }

        max_alt
    }
}
